src/main/java/com/cdancy/jenkins/rest/JenkinsClient.java

```diff
@@ -49,29 +55,46 @@ public JenkinsClient() {
      * @param endPoint URL of Jenkins instance.
      * @param authentication authentication used to connect to Jenkins instance.
      * @param overrides jclouds Properties to override defaults when creating a new JenkinsApi.
+     * @param modules a list of modules to be passed to the Contextbuilder, e.g. for logging.
      */
     public JenkinsClient(@Nullable final String endPoint,
             @Nullable final JenkinsAuthentication authentication,
-            @Nullable final Properties overrides) {
+            @Nullable final Properties overrides,
+            @Nullable final List<Module> modules) {
         this.endPoint = endPoint != null
                 ? endPoint
                 : JenkinsUtils.inferEndpoint();
         this.credentials = authentication != null
                 ? authentication
                 : JenkinsUtils.inferAuthentication();
         this.overrides = mergeOverrides(overrides);
-        this.jenkinsApi = createApi(this.endPoint, this.credentials, this.overrides);
+        this.modules = addModules(this.credentials, modules);
+        this.jenkinsApi = createApi(this.endPoint, this.credentials, this.overrides, this.modules);
     }
 
-    private JenkinsApi createApi(final String endPoint, final JenkinsAuthentication authentication, final Properties overrides) {
+    private JenkinsApi createApi(final String endPoint, final JenkinsAuthentication authentication, final Properties overrides, final List<Module> modules) {
         return ContextBuilder
                 .newBuilder(new JenkinsApiMetadata.Builder().build())
                 .endpoint(endPoint)
-                .modules(Lists.newArrayList(new JenkinsAuthenticationModule(authentication)))
+                .modules(modules)
                 .overrides(overrides)
                 .buildApi(JenkinsApi.class);
     }
 
+    /**
+     * Add the defaul built-in module, followed by the user modules.
+     *
+     * @param modules a list of user provided modules.
+     * @return The complete list of modules starting with the default ones.
+     */
+    private List<Module> addModules(final JenkinsAuthentication authentication, final List<Module> modules) {
+        final List<Module> allModules = Lists.newArrayList(new JenkinsAuthenticationModule(authentication));
+        if (modules != null) {
+            allModules.addAll(modules);
+        }
+        return allModules;
+    }
```

Comment:
One last comment: going to push back against this here as well and instead lets keep this code within the `createApi` method. The code itself is fine but only makes sense if we are exposing the modules which I don't think we have a good reason for doing at the moment.